File: hackerrank/easy/0043-staircase/js/solution.mjs
// solution.mjs

export function staircase(n) {
  if (n <= 0) return;
  for (let i = 1; i <= n; i++) {
    const spaces = " ".repeat(n - i);
    const hashes = "#".repeat(i);
    console.log(spaces + hashes);
  }
}

// --- demo run ---
staircase(6);
/*
Expected:
     #
    ##
   ###
  ####
 #####
######
*/

File: hackerrank/easy/0043-staircase/swift/solution.swift
// solution.swift

import Foundation

func staircase(n: Int) {
    guard n > 0 else { return }
    for i in 1...n {
        let spaces = String(repeating: " ", count: n - i)
        let hashes = String(repeating: "#", count: i)
        print(spaces + hashes)
    }
}

// --- demo run ---
staircase(n: 6)
/*
Expected:
     #
    ##
   ###
  ####
 #####
######
*/